controller/instance_manager_controller.go

```diff
@@ -468,26 +468,60 @@ func (imc *InstanceManagerController) syncInstanceManagerPDB(im *longhorn.Instan
 		return err
 	}
 
+	clusterAutoscalerEnabled, err := imc.ds.GetSettingAsBool(types.SettingNameKubernetesClusterAutoscalerEnabled)
+	if err != nil {
+		return err
+	}
+
 	// When current node is unschedulable, it is a signal that the node is being cordoned/drained.
 	if unschedulable {
 		if imPDB == nil {
 			return nil
 		}
-		canDelete, err := imc.canDeleteInstanceManagerPDB(im)
+
+		imc.logger.Debugf("Node %v is marked unschedulable, try to remove %v PBD", imc.controllerID, im.Name)
+		canDelete, availableNode, err := imc.canDeleteInstanceManagerPDB(im)
 		if err != nil {
 			return err
 		}
+		if clusterAutoscalerEnabled {
+			if !canDelete && availableNode == "" {
+				return errors.Errorf("cannot find healthy replica on another node")
+			}
+
+			node, err := imc.ds.GetNode(availableNode)
+			if err != nil {
+				return err
+			}
+			imc.logger.Debugf("Creating instance manager PBD on %v", node.Name)
+			imc.enqueueKubernetesNode(node)
+			imc.logger.Debugf("Deleting %v PDB on %v", im.Name, imc.controllerID)
+			return imc.deleteInstanceManagerPDB(im)
```

Comment:
We should not just `deleteInstanceManagerPDB` unless the PDB has been created in the available node.

Reply:
Fixed, and added case to test cordon node [here](https://github.com/longhorn/longhorn-tests/pull/936/commits).